pyomo/contrib/doe/fim_doe.py

```diff
@@ -1115,15 +1387,24 @@ def jac_numerical(m,j,p,t):
             '''
             # A better way to do this: 
             # https://github.com/IDAES/idaes-pse/blob/274e58bef55f2f969f0df97cbb1fb7d99342388e/idaes/apps/uncertainty_propagation/sens.py#L296
-            if self.measurement_extra_index[j] is None:
+            # check if j is a measurement with extra index by checking if there is '_index_' in its name
+            if '_index_' in j:
+                measure_name = j.split('_index_')[0]
+                measure_index = j.split('_index_')[1]
+                if type(self.measure.name_and_index[measure_name][0]) is str:
+                    measure_index = '"' + measure_index + '"'
+                if t in self.flatten_measure_timeset[j]:
+                    up_C = eval('m.' + measure_name + '[' + str(scenario_all['jac-index'][p][0]) + ',' + measure_index + ',' + str(t) + ']')
```

Comment:
Can the measurement class give us the Pyomo string name? Here we would just provide `str(scenario_all['jac-index'][p][0])` as an input to a function in the measurement class.